File: Frontend/nimrag-frontend/src/types/market.ts
export interface MarketItem {
  id: string
  symbol: string
  name: string
  price: number
  change: number
  percent_change: number
  currency: string
  asset_type: 'stock' | 'crypto'
}

export interface MarketResponse {
  items: MarketItem[]
  source: 'live' | 'cache'
}

File: Backend/src/schemas/market.py
from typing import Literal

from pydantic import BaseModel, Field


class MarketItem(BaseModel):
    id: str
    symbol: str
    name: str
    price: float
    change: float
    percent_change: float
    currency: str
    asset_type: Literal["stock", "crypto"]


class MarketResponse(BaseModel):
    items: list[MarketItem] = Field(default_factory=list)
    source: Literal["live", "cache"] = "live"
